core/src/display_object/movie_clip.rs

```diff
@@ -765,19 +777,17 @@ impl<'gc> MovieClip<'gc> {
         // Clamp frame number in bounds.
         let frame = frame.max(1);
 
-        if frame != self.current_frame() {
```

Comment:
we now `run_goto` for same-frame gotos even in avm1? Just making sure

Reply:
This is actually incorrect, at least for AVM1, and adding the check back there fixes Bloxorz. Surprised it didn't trip any regression tests, but I'll add an AvmType check here...

For those wondering, in AVM2, we actually do have to allow running redundant gotos. Running `this.gotoAndStop(1)` *on* frame 1 causes frame scripts to queue and run before the goto returns. There's two tests in this PR that will fail if we don't do this.